pkg/registry/apis/query/query.go

```diff
@@ -160,21 +160,17 @@ func (r *queryREST) Connect(connectCtx context.Context, name string, _ runtime.O
 		// Actually run the query
 		rsp, err := b.execute(ctx, req)
 		if err != nil {
-			// log unexpected errors
-			var k8sErr *errorsK8s.StatusError
-			if errors.As(err, &k8sErr) {
-				// we do not need to log 4xx errors as they are expected
-				if k8sErr.ErrStatus.Code >= 500 {
-					b.log.Error("hit unexpected k8s error while executing query", "err", err, "status", k8sErr.Status())
-				}
-				b.log.Debug("sending a known k8s error to the client", "err", err, "status", k8sErr.Status())
+			b.log.Error("execute error", "http code", query.GetResponseCode(rsp), "err", err)
```

Comment:
if you don't have a rsp will it error out?

Reply:
maybe something like if rsp exists use that status code otherwise it's 500

Reply:
if `rsp` is nil, `GetResponseCode` fills in `400`. I'll tweak how `GetResponseCode` works. Good shout!